data_platform/ingestion/sync_checkpoint.py

```
@@ -383,6 +383,112 @@ def ensure_dataset_manifest(
+        New run directory and the flushed metadata payload.
+
+    Raises
+    ------
+    ValueError
+        When an unfinished raw run already exists for this dataset.
+    """
+    unfinished_dir = find_resume_run_dir(storage, run_dir_name=None)
+    if unfinished_dir is not None:
+        raise ValueError(
+            f"An unfinished raw run exists at {unfinished_dir}; resume it instead of starting a new run"
+        )
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)